issue_links.go

```diff
@@ -0,0 +1,181 @@
+	"net/url"
+	"time"
+)
+
+// IssueLinksService handles communication with the issue relations related methods
+// of the GitLab API.
+//
+// GitLab API docs: https://docs.gitlab.com/ee/api/issue_links.html
+type IssueLinksService struct {
+	client *Client
+}
+
+// IssueLinks represents a GitLab IssueLink.
+//
+// GitLab API docs: https://docs.gitlab.com/ee/api/issue_links.html
+type IssueLinks struct {
+	ID          int        `json:"id,omitempty"`
+	IID         int        `json:"iid,omitempty"`
+	ProjectID   int        `json:"project_id,omitempty"`
+	Title       string     `json:"title,omitempty"`
+	Description string     `json:"description,omitempty"`
+	State       string     `json:"state,omitempty"`
+	UpdatedAt   *time.Time `json:"updated_at,omitempty"`
+	CreatedAt   *time.Time `json:"created_at,omitempty"`
+	ClosedAt    *time.Time `json:"closed_at,omitempty"`
+	Labels      []string   `json:"labels,omitempty"`
+	Milestone   *Milestone `json:"milestone,omitempty"`
+	Assignees   []struct {
+		ID        int    `json:"id,omitempty"`
+		Name      string `json:"name,omitempty"`
+		Username  string `json:"username,omitempty"`
+		State     string `json:"state,omitempty"`
+		AvatarURL string `json:"avatar_url,omitempty"`
+		WebURL    string `json:"web_url,omitempty"`
+	} `json:"assignees,omitempty"`
+	Author struct {
+		ID        int    `json:"id,omitempty"`
+		Name      string `json:"name,omitempty"`
+		Username  string `json:"username,omitempty"`
+		State     string `json:"state,omitempty"`
+		AvatarURL string `json:"avatar_url,omitempty"`
+		WebURL    string `json:"web_url,omitempty"`
+	} `json:"author,omitempty"`
+	Assignee struct {
+		ID        int    `json:"id,omitempty"`
+		Name      string `json:"name,omitempty"`
+		Username  string `json:"username,omitempty"`
+		State     string `json:"state,omitempty"`
+		AvatarURL string `json:"avatar_url,omitempty"`
+		WebURL    string `json:"web_url,omitempty"`
+	} `json:"assignee,omitempty"`
+	UserNotesCount   int        `json:"user_notes_count,omitempty"`
+	Upvotes          int        `json:"upvotes,omitempty"`
+	Downvotes        int        `json:"downvotes,omitempty"`
+	DueDate          *ISOTime   `json:"due_date,omitempty"`
+	Confidential     bool       `json:"confidential,omitempty"`
+	Weight           int        `json:"weight,omitempty"`
+	DiscussionLocked bool       `json:"discussion_locked,omitempty"`
+	WebURL           string     `json:"web_url,omitempty"`
+	TimeStats        *TimeStats `json:"time_stats,omitempty"`
+	Links            struct {
+		Self       string `json:"self,omitempty"`
+		Notes      string `json:"notes,omitempty"`
+		AwardEmoji string `json:"award_emoji,omitempty"`
+		Project    string `json:"project,omitempty"`
+	} `json:"_links,omitempty"`
+	Subscribed  bool `json:"subscribed,omitempty"`
+	IssueLinkID int  `json:"issue_link_id,omitempty"`
+}
+
+type IssueLinksResult struct {
+	SourceIssue *IssueLinks `json:"source_issue"`
+	TargetIssue *IssueLinks `json:"target_issue"`
+}
```

Comment:
Is this really how the output is given? As the docs seem to indicate that this call would return a list of issue links related to the given issue.

Reply:
Yes, I tested it on my end it's only used for the `POST` and `DELETE` endpoints. The `GET` endpoints returns a list of `IssueLink`

Reply:
Ok... I see it as well now when looking at the docs of the `POST` and `DELETE` calls again.